pkg/discovery/network/openshift4.go

```diff
@@ -49,7 +49,27 @@ func discoverOpenShift4Network(ctx context.Context, client controllerClient.Clie
 		return nil, errors.WithMessage(err, "error obtaining the default 'cluster' OpenShift4 Network config resource")
 	}
 
-	return parseOS4Network(network)
+	clusterNetwork, err := parseOS4Network(network)
+	if err != nil {
+		return nil, err
+	}
+
+	if clusterNetwork.NetworkPlugin == cni.OVNKubernetes {
+		ovnDBPod, err := FindPod(ctx, client, "name=ovnkube-db")
+		if err != nil {
+			return nil, err
+		}
+
+		if ovnDBPod == nil {
+			// IC is enabled and Openshift uses zone per node and to connect to OVN DB using
+			// socket connection we need use this path.
```

Comment:
While you are addressing the commit message, you can update this nit as well.

```suggestion
			// socket connection we need to use this path.
```

Reply:
Done.